src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.cs

```diff
@@ -42,9 +41,6 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             context.RegisterSourceOutput(inputData, (spc, source) => Execute(source.Item1, source.Item2, spc));
```

Comment:
The equals implementation on `BinderInvocation` uses ref equality, not value equality. That means the `CreateSyntaxProvider` call above is never cached. This is an issue for IDE performance.

Reply:
I'd say this fact pre-dates this PR, so I'll address outside of it, for GA if necessary - https://github.com/dotnet/runtime/issues/83534#issuecomment-1690776940.

We've made related perf fixes, with metrics indicating that IDE perf is [at par](https://gist.github.com/layomia/5115262732865825cfba565e71078271) with other generators https://github.com/dotnet/runtime/pull/90746 https://github.com/dotnet/runtime/pull/89266 wrt average time spent in the generator, testing with the Roslyn repo as a "large" solution.

So I believe we're fine for GA. We have some other improvements lined up including making the generator incremental (https://github.com/dotnet/runtime/pull/89587). Further constraining candidate invocations in the initial shallow check for valid calls to intercept might be impactful & we could pull it into the release if necessary - https://github.com/dotnet/runtime/issues/9068.

Reply:
> with metrics indicating that IDE perf is [at par](https://gist.github.com/layomia/5115262732865825cfba565e71078271) with other generators

Do these benchmarks measure incremental performance?

Reply:
If the most important aspect is to exercise the parsing codepaths multiple times as a result of multiple edits, then yes.

I enabled the generator in all the projects in the Roslyn solution and made repeated edits in a few of them. The perf numbers reflect the aggregate of this activity. The generator parses all invocations but doesn't emit anything (these projects don't do binding). I consider this a sanity-check stress test (i.e in a large solution) that the enabled generator isn't generally detrimental to IDE perf. A more probable instance of this scenario is an AOT'd Web SDK app (where both RDG and config binding generators are enabled by default) that doesn't do any config binding.

The test doesn't cover the current behavior of re-emitting logic in projects that actually perform binding operations. In common cases there are only a few invocations and target config types, and I estimate that emitter logic is far less expensive than the parser's. I'll take measurements to see if we need any improvement for this scenario in RC 2.

Ultimately we've described workarounds for IDE perf issues here https://github.com/dotnet/runtime/issues/83534#issuecomment-1679382605. We can could document them in release notes if needed.

Reply:
Is this generator alwasy on or only conditionally on?

Reply:
It's conditionally on -- on by default in NativeAOT web sdk apps, otherwise off.